### ISSUE TITLE
doesnt run on M1 mac

### ISSUE THREAD

**Issue by halpz** [jef/streetmerchant#2639]
npm ERR! code 1
npm ERR! path /Users/halpz/Desktop/streetmerchant/streetmerchant/node_modules/puppeteer-extra-plugin-adblocker/node_modules/puppeteer
npm ERR! command failed
npm ERR! command sh -c node install.js
npm ERR! The chromium binary is not available for arm64: 
npm ERR! If you are on Ubuntu, you can install with: 
npm ERR! 
npm ERR!  apt-get install chromium-browser
npm ERR! 
npm ERR! /Users/halpz/Desktop/streetmerchant/streetmerchant/node_modules/puppeteer-extra-plugin-adblocker/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserFetcher.js:112
npm ERR!             throw new Error();
npm ERR!             ^
npm ERR! 
npm ERR! Error
npm ERR!     at /Users/halpz/Desktop/streetmerchant/streetmerchant/node_modules/puppeteer-extra-plugin-adblocker/node_modules/puppeteer/lib/cjs/puppeteer/node/BrowserFetcher.js:112:19
npm ERR!     at FSReqCallback.oncomplete (node:fs:195:21)

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/halpz/.npm/_logs/2021-06-21T15_23_51_443Z-debug.log

**Comment by meta-434** [comment]
yeah, this is a Puppeteer thing, not a streetmerchant thing. Apparently you can work around it, though YMMV: https://github.com/puppeteer/puppeteer/issues/6622

**Comment by joshuaali** [comment]
The following worked for me:

1. Tell Puppeteer to not download Chromium
   *  Temporarily: `export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1`
   *  Permanently: `npm config set puppeteer_skip_chromium_download 1`
2. `npm i`
3. `brew install chromium`
   * I had to also clear extended attributes via `xattr -c /Applications/Chromium.app`
4. `echo PUPPETEER_EXECUTABLE_PATH="$(which chromium) >> dotenv`
5. `npm start`

**Comment by ophello** [comment]
> The following worked for me:
> 
> 1. Tell Puppeteer to not download Chromium
>    
>    * Temporarily: `export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1`
>    * Permanently: `npm config set puppeteer_skip_chromium_download 1`
> 2. `npm i`
> 3. `brew install chromium`
>    
>    * I had to also clear extended attributes via `xattr -c /Applications/Chromium.app`
> 4. `echo PUPPETEER_EXECUTABLE_PATH=$(which chromium) >> dotenv`
> 5. `npm start`

I don't understand these instructions. Can you dumb them down for me? Where are you changing these settings and in what files? I tried this and it completely failed.

**Comment by ophello** [comment]
> The following worked for me:
> 
> 1. Tell Puppeteer to not download Chromium
>    
>    * Temporarily: `export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=1`
>    * Permanently: `npm config set puppeteer_skip_chromium_download 1`
> 2. `npm i`
> 3. `brew install chromium`
>    
>    * I had to also clear extended attributes via `xattr -c /Applications/Chromium.app`
> 4. `echo PUPPETEER_EXECUTABLE_PATH=$(which chromium) >> dotenv`
> 5. `npm start`

I tried this and it failed. Does it matter what directory I'm in when doing these steps in the Terminal?

**Comment by joshuaali** [comment]
Steps 2, 4, and 5 assumes that you're in the directory in which you've checked out the project.

**Comment by ophello** [comment]
> Steps 2, 4, and 5 assumes that you're in the directory in which you've checked out the project.

And at what stage during the installation of streetmerchant are these steps supposed to occur? I'm worried that doing stuff out of order is causing problems. I'm hoping for a comprehensive installation process that includes these fixes, rather than fixing after the fact.

**Comment by meta-434** [comment]
@ophello 

after `git clone https://github.com/jef/streetmerchant.git` and `cd streetmerchant`.

**Comment by ophello** [comment]
I was able to get this to work by making a few changes before installing streetmerchant:

```
brew install --no-quarantine chromium
```
This installs Chromium without a quarantine in later versions of OSX. This was the problem that kept causing everything to fail.

Then we make a new .zshrc file:
```
sudo nano ~/.zshrc
```
Then set the file contents to this:
```
export PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
export PUPPETEER_EXECUTABLE_PATH=$(which chromium)
```

Success!
